bundle/edu.gemini.p2checker/src/test/scala/edu/gemini/p2checker/rules/RuleSpec.scala

```diff
@@ -0,0 +1,56 @@
+package edu.gemini.p2checker.rules
+
+import java.util.UUID
+
+import edu.gemini.p2checker.api.{IRule, ObservationElements}
+import edu.gemini.pot.sp.{ISPObservation, SPComponentType}
+import edu.gemini.pot.util.POTUtil
+import edu.gemini.spModel.core.SPProgramID
+import edu.gemini.spModel.data.ISPDataObject
+import edu.gemini.spModel.rich.pot.sp._
+import edu.gemini.spModel.target.obsComp.TargetObsComp
+import org.specs2.mutable.Specification
+
+import scala.collection.JavaConverters._
+import scalaz.syntax.id._
+
+/**
+ * Generic test bed for arbitrary P2 checker rule tests.
+ */
+abstract class RuleSpec extends Specification {
```

Comment:
Many, see `AbstractRuleTest`, `GeneralRuleTest` and `GmosRuleTest` for Java and then there are more written in Scala. I was tempted to start moving some from `GmosRuleTest`, I think with this approach we could probably have all those tests with about 1/4 of the code, but in the end there are only so many hours in a day.